lib/swim.ml

```diff
@@ -0,0 +1,127 @@
+open Base.Hashtbl
+
+type message =
+  | Ping
+  | Acknowledge
+  | PingRequest of Peer.address
+[@@deriving bin_io]
+
+type config = {
+  protocol_period : int;
+  round_trip_time : int;
+  timeout : int;
+}
+
+type t = {
```

Comment:
`Swim.t` does not say a lot about what this object actually is. Maybe we can bypass the use of SWIM in the name and just call this module "Failure_detector" as in the paper. That way, `Failure_detector.t`'s purpose and functionality is more self-explanatory even if the consumer/reader of the code is unfamiliar with SWIM.

Reply:
Totally agreed, renamed to `Failure_detector`!